src/solvers/refinement/string_refinement.cpp

```diff
@@ -65,6 +66,52 @@ void string_refinementt::set_mode()
 
 /*******************************************************************\
 
+Function: string_refinementt::set_max_string_length
+
+  Inputs:
+    i - maximum length which is allowed for strings.
```

Comment:
I'd prefer using an unsigned integer and std::numeric_limits<unsigned>::max() as default upper limit. What is "no limit" supposed to mean in your case? What is the relation to MAX_CONCRETE_STRING_SIZE?